### PR TITLE
[Impeller] re-enable gold CTL.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Add goldctl dependency to arm64

Fixes https://github.com/flutter/flutter/issues/147180

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Also includes a revert of https://github.com/flutter/engine/pull/52137 which can't be autoreverted.

**Comment by jonahwilliams** [comment]
And including https://github.com/flutter/engine/pull/52106 too

**Comment by jonahwilliams** [comment]
Also reverting https://github.com/flutter/engine/pull/52180

**Comment by jonahwilliams** [comment]
Adding https://github.com/flutter/engine/pull/52160 speculative revert

**Review by matanlurey — APPROVED** [review]

File: impeller/aiks/aiks_unittests.cc [hunk 1]
```diff
@@ -2348,6 +2349,10 @@ TEST_P(AiksTest, DrawPaintTransformsBounds) {
 }
 
 TEST_P(AiksTest, CanDrawPoints) {
+  if (GetBackend() == PlaygroundBackend::kMetal) {
+    // https://github.com/flutter/flutter/issues/147184
```
Comment (gaaclarke) [inline impeller/aiks/aiks_unittests.cc:2353]: Weird.

**Review by gaaclarke — APPROVED** [review]
This approach sounds good to me, I'm not sure if we'll see more problems but looks good to me as it is now.

**Pull request merged**